Index: Graphics/SpriteBatch.cpp
===================================================================
--- Graphics/SpriteBatch.cpp	(revision 365)
+++ Graphics/SpriteBatch.cpp	(working copy)
@@ -373,10 +373,16 @@
 	void SpriteBatch::DrawString(SpriteFont* spriteFont, const wchar_t* text, const Vector2& position, Color color,
 		float rotation, const Vector2& origin, float scale, SpriteEffects effects, float layerDepth)
 	{
+		DrawString(spriteFont, text, position, color, rotation, origin, Nxna::Vector2(scale, scale), effects, layerDepth);
+	}
+
+	void SpriteBatch::DrawString(SpriteFont* spriteFont, const wchar_t* text, const Vector2& position, Color color,
+		float rotation, const Vector2& origin, const Nxna::Vector2& scale, SpriteEffects effects, float layerDepth)
+	{
 		Matrix rotationMatrix;
 		Matrix::CreateRotationZ(rotation, rotationMatrix);
 		Matrix translationMatrix;
-		Matrix::CreateTranslation(-origin.X * scale, -origin.Y * scale, 0, translationMatrix);
+		Matrix::CreateTranslation(-origin.X * scale.X, -origin.Y * scale.Y, 0, translationMatrix);
 		Matrix transform;
 		Matrix::Multiply(translationMatrix, rotationMatrix, transform);
 
@@ -395,18 +401,18 @@
 			spriteFont->GetCharacterInfo(c, &glyph, &cropping, &kerning);
 
 			if (!ignoreSpacing)
-				cursor.X += spriteFont->GetSpacing() * scale;
+				cursor.X += spriteFont->GetSpacing() * scale.X;
 
-			cursor.X += kerning.X * scale;
+			cursor.X += kerning.X * scale.X;
 
 			Vector2 transformedPosition = cursor;
-			transformedPosition.X += cropping.X * scale;
-			transformedPosition.Y += cropping.Y * scale;
+			transformedPosition.X += cropping.X * scale.X;
+			transformedPosition.Y += cropping.Y * scale.Y;
 			Nxna::Vector2::Transform(transformedPosition, transform, transformedPosition);
 
 			Draw(spriteFont->m_texture, transformedPosition + position, &glyph, color, rotation, Vector2(0, 0), scale, effects, layerDepth);
 
-			cursor.X += (kerning.Y + kerning.Z) * scale;
+			cursor.X += (kerning.Y + kerning.Z) * scale.X;
 
 			ignoreSpacing = false;
 		}
Index: Graphics/SpriteBatch.h
===================================================================
--- Graphics/SpriteBatch.h	(revision 365)
+++ Graphics/SpriteBatch.h	(working copy)
@@ -93,6 +93,8 @@
 			float rotation, const Vector2& origin, float scale, SpriteEffects effects, float layerDepth);
 		void DrawString(SpriteFont* spriteFont, const wchar_t* text, size_t numCharacters, const Vector2& position, Color color,
 			float rotation, const Vector2& origin, float scale, SpriteEffects effects, float layerDepth);
+		void DrawString(SpriteFont* spriteFont, const wchar_t* text, const Vector2& position, Color color,
+			float rotation, const Vector2& origin, const Nxna::Vector2& scale, SpriteEffects effects, float layerDepth);
 		void DrawString(SpriteFont* spriteFont, const char* text, const Vector2& position, Color color);
 		void DrawString(SpriteFont* spriteFont, const char* text, const Vector2& position, Color color,
 			float rotation, const Vector2& origin, float scale, SpriteEffects effects, float layerDepth);
